scripts/src/utils/RateLimit.ts

```diff
@@ -0,0 +1,75 @@
+import * as moment from "moment";
+import { getRedisClient, RedisAsyncClient } from "../redis";
+import { TooManyRegistrations, TooMuchEarnOrdered } from "../errors";
+
+class RateLimit {
+	private readonly bucketPrefix: string;
+	private readonly rateLimitValue: number;
+	private readonly redis: RedisAsyncClient;
+	private windowSize: number = 0;
+	private ttl: number = 0;
+	private readonly ttlWindowRatio: number = 10;
+
+	constructor(bucketPrefix: string, rateLimitValue: number, windowSizeMomentObject: moment.Duration) {
+		this.bucketPrefix = bucketPrefix;
+		this.rateLimitValue = rateLimitValue;
+		this.windowSize = windowSizeMomentObject.asSeconds();
+		this.ttl = this.ttl * this.ttlWindowRatio;
+		this.redis = getRedisClient();
+	}
+
+	public async checkRate(): Promise<boolean> {
+		return await this.check(1);
+	}
+
+	public async checkAmount(amount: number): Promise<boolean> {
+		return await this.check(amount);
+	}
+
+	private async check(step: number) {
+		const bucketSize = Math.max(this.windowSize / 60, 1); // resolution
+		const currentTimestampSeconds = Math.floor((Date.now() / 1000) / bucketSize) * bucketSize; // int(currentTimestampSeconds / 60) * 60
+		const currentBucketName = this.bucketPrefix + currentTimestampSeconds;
+
+		await this.redis.async.incrby(currentBucketName, step);
+		this.redis.expire(currentBucketName, this.ttl);
+
+		const windowKeys: string[] = [currentTimestampSeconds.toString()];
```

Comment:
`const windowKeys: string[] = []`
I don't understand why you need the first line. Plus it's wrong because it's missing the prefix

Reply:
Prefix added, missed it :(
I need windowKeys for mget asking: it stores all possible keys

Reply:
You don't need the first element prepopulated